In [1]:
import numpy as np

In [2]:
import matplotlib.pyplot as plt

In [3]:
import seaborn as sns

In [4]:
 import pandas as pd  

In [5]:
gdp_df=pd.read_csv('../data/gdp_percapita.csv')

In [6]:
print(gdp_df)

     Country or Area  Year        Value  Value Footnotes
0        Afghanistan  2021  1517.016266              NaN
1        Afghanistan  2020  1968.341002              NaN
2        Afghanistan  2019  2079.921861              NaN
3        Afghanistan  2018  2060.698973              NaN
4        Afghanistan  2017  2096.093111              NaN
...              ...   ...          ...              ...
7657        Zimbabwe  1994  2670.106615              NaN
7658        Zimbabwe  1993  2458.783255              NaN
7659        Zimbabwe  1992  2468.278257              NaN
7660        Zimbabwe  1991  2781.787843              NaN
7661        Zimbabwe  1990  2704.757299              NaN

[7662 rows x 4 columns]


In [7]:
print(gdp_df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7662 entries, 0 to 7661
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Country or Area  7662 non-null   object 
 1   Year             7662 non-null   int64  
 2   Value            7662 non-null   float64
 3   Value Footnotes  0 non-null      float64
dtypes: float64(2), int64(1), object(1)
memory usage: 239.6+ KB
None


How many rows and columns are in `gdp_df`? 7662 rows x 4 columns
What are the data types of each column? object, int64, float64

In [8]:
gdp_df.drop(gdp_df.columns[[3]], axis=1, inplace=True)

In [9]:
print(gdp_df)

     Country or Area  Year        Value
0        Afghanistan  2021  1517.016266
1        Afghanistan  2020  1968.341002
2        Afghanistan  2019  2079.921861
3        Afghanistan  2018  2060.698973
4        Afghanistan  2017  2096.093111
...              ...   ...          ...
7657        Zimbabwe  1994  2670.106615
7658        Zimbabwe  1993  2458.783255
7659        Zimbabwe  1992  2468.278257
7660        Zimbabwe  1991  2781.787843
7661        Zimbabwe  1990  2704.757299

[7662 rows x 3 columns]


In [10]:
gdp_df.rename(columns={"Country or Area": "Country", "Value": "GDP_Per_Capita"})

Country  Year  GDP_Per_Capita
0     Afghanistan  2021     1517.016266
1     Afghanistan  2020     1968.341002
2     Afghanistan  2019     2079.921861
3     Afghanistan  2018     2060.698973
4     Afghanistan  2017     2096.093111
...           ...   ...             ...
7657     Zimbabwe  1994     2670.106615
7658     Zimbabwe  1993     2458.783255
7659     Zimbabwe  1992     2468.278257
7660     Zimbabwe  1991     2781.787843
7661     Zimbabwe  1990     2704.757299

[7662 rows x 3 columns]

In [11]:
unique_country=gdp_df.drop_duplicates(subset='Country or Area')

In [12]:
print(unique_country)

                  Country or Area  Year         Value
0                     Afghanistan  2021   1517.016266
20    Africa Eastern and Southern  2022   3553.913370
53     Africa Western and Central  2022   4063.857691
86                        Albania  2022  15501.662931
119                       Algeria  2022  11187.382303
...                           ...   ...           ...
7501                     Viet Nam  2022  11396.531469
7534           West Bank and Gaza  2022   5722.409175
7563                        World  2022  17485.934316
7596                       Zambia  2022   3298.142890
7629                     Zimbabwe  2022   2143.237574

[242 rows x 3 columns]


In [13]:
plt.hist(gdp_df.value)

AttributeError: 'DataFrame' object has no attribute 'value'

In [12]:
plt.show()

In [16]:
plt.gaussian_kde(gdp_df.value)

AttributeError: module 'matplotlib.pyplot' has no attribute 'gaussian_kde'